Validation improvements

TW80000 opened the pull request:
- If a property on a model has a `validateObj` function, then use that to validate the UI input when that property is used as a viewspec.
- If the value in the ui element is invalid, show a popup with a message returned by the `validateObj` function. Also change the border colour to red to indicate that the value is invalid.

Before entering any text:
<img width="981" alt="screen shot 2018-09-07 at 1 48 00 pm" src="https://user-images.githubusercontent.com/4259165/45235763-f08f3600-b2a7-11e8-8d2f-3134c83bad2d.png">

After entering invalid text and changing focus away from the input:
<img width="991" alt="screen shot 2018-09-07 at 1 48 11 pm" src="https://user-images.githubusercontent.com/4259165/45235761-f08f3600-b2a7-11e8-8f7a-478f90e3add7.png">

After clicking anywhere on the screen to hide the popup, border stays red to show invalid state:
<img width="983" alt="screen shot 2018-09-07 at 1 48 20 pm" src="https://user-images.githubusercontent.com/4259165/45235760-f08f3600-b2a7-11e8-902d-c0fcd4a09dc5.png">

Part of the solution to https://github.com/nanoPayinc/NANOPAY/issues/3451

Commit:
Don't allow legal name to be edited directly

Commit:
Improve style of PopupView

Commit:
Fix conditions

Checking for truthiness treats 0 as false, but 0 is a valid value to set
a numeric property to.

Commit:
Code cleanup

Commit:
Add validation to inputs

kgrgreer commented:
Will review.

zahnno commented on src/foam/u2/tag/Input.js:
```diff
@@ -56,7 +82,22 @@ foam.CLASS({
+        'validateObj' function on a property. Doing that will automatically set
+        this property to the validateObj function you defined on the property.
+      `,
+      value: function() {}
```
Add a console error if the value being set isn't a function.

zahnno commented on src/foam/u2/tag/Input.js:
```diff
@@ -32,6 +45,19 @@ foam.CLASS({
           return o;
         }
+
+        if ( typeof this.validateValue === 'function' && o != undefined ) {
```
validateValue can always equate a function if you make it a requirement on the prop.

Commit:
Make assert message more informative

Commit:
Set proper class on validateValue

Also fixes a naming mistake. I had named the property validationCheck
but was setting and using validateValue in the code. This commit makes
the naming consistent and set the property's type to 'Function', which
gives us value checking on set and default value out of the box.

TW80000 commented:
All comments addressed. Ready for review again.

TW80000 commented:
Closing because I don't think we want this anymore.
